src/plugins/workspace/public/plugin.ts

```diff
@@ -318,14 +319,26 @@ export class WorkspacePlugin implements Plugin<{}, {}, WorkspacePluginSetupDeps>
     core.application.register({
       id: WORKSPACE_LIST_APP_ID,
       title: '',
-      navLinkStatus: AppNavLinkStatus.hidden,
+      // nav link should be visible when nav group enabled
+      navLinkStatus: core.chrome.navGroup.getNavGroupEnabled()
+        ? AppNavLinkStatus.visible
+        : AppNavLinkStatus.hidden,
```

Comment:
can we add a comment of when core.chrome.navGroup.getNavGroupEnabled() changed, the page will refresh and application setup will execute it again?